File: ProyectoMysql/server/EntityLayer/OrdenPedidoDetalleEntity.py
from datetime import datetime
from pydantic import BaseModel
from Utilidades.Enumerado.ProcessActionEnum import ProcessActionEnum


class OrdenPedidoDetalleSaveModel(BaseModel):
    OrdenPedidoDetalleId: int 
    OrdenPedidoId: int 
    MercaderiaId: int 
    UnidadMedidaId: int 
    CantidadSolicitado: float 
    CantidadReservado: float 
    CantidadFaltante: float 
    CantidadAtendido: float 
    Enlazado: bool 
    Atendido: bool 
    Action: ProcessActionEnum


class OrdenPedidoDetalleItemModel:
    OrdenPedidoDetalleId: int 
    OrdenPedidoId: int 
    MercaderiaId: int 
    UnidadMedidaId: int 
    CantidadSolicitado: float 
    CantidadReservado: float 
    CantidadFaltante: float 
    CantidadAtendido: float 
    Enlazado: bool 
    Atendido: bool 
    NomProducto: str
    CategoriaId: int
    CodigoUM: str
    Stock: float
    # Guid: str

    def Cargar(_DB):
        c = OrdenPedidoDetalleItemModel()
        c.OrdenPedidoDetalleId = _DB["OrdenPedidoDetalleId"]
        c.OrdenPedidoId = _DB["OrdenPedidoId"]
        c.MercaderiaId = _DB["MercaderiaId"]
        c.UnidadMedidaId = _DB["UnidadMedidaId"]
        c.CantidadSolicitado = _DB["CantidadSolicitado"]
        c.CantidadReservado = _DB["CantidadReservado"]
        c.CantidadFaltante = _DB["CantidadFaltante"]
        c.CantidadAtendido = _DB["CantidadAtendido"]
        c.Enlazado = bool(ord(_DB["Enlazado"]))
        c.Atendido = bool(ord(_DB["Atendido"]))
        return c

    def CargarCabecera(_DB):
        c = OrdenPedidoDetalleItemModel()
        c.OrdenPedidoDetalleId = _DB["OrdenPedidoDetalleId"]
        c.OrdenPedidoId = _DB["OrdenPedidoId"]
        c.MercaderiaId = _DB["MercaderiaId"]
        c.UnidadMedidaId = _DB["UnidadMedidaId"]
        c.CantidadSolicitado = _DB["CantidadSolicitado"]
        c.CantidadReservado = _DB["CantidadReservado"]
        c.CantidadFaltante = _DB["CantidadFaltante"]
        c.CantidadAtendido = _DB["CantidadAtendido"]
        c.Enlazado = bool(ord(_DB["Enlazado"]))
        c.Atendido = bool(ord(_DB["Atendido"]))
        c.NomProducto = _DB["NomProducto"]
        c.CategoriaId = _DB["CategoriaId"]
        c.CodigoUM = _DB["CodigoUM"]
        c.Stock = _DB["Stock"]
        # c.Guid=''
        return c

File: ProyectoMysql/Api/LogisticStorage/LogisticStorage.EntityLayer/OrdenPedidoDetalleEntity.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace LogisticStorage.EntityLayer
{
    public partial class OrdenPedidoDetalleEntity : BaseEntityObject
    {
        public OrdenPedidoDetalleEntity()
        {
            this.OrdenPedidoDetalleId = 0;
            this.OrdenPedidoId = 0;
            this.MercaderiaId = 0;
            this.UnidadMedidaId = 0;
            this.CantidadSolicitado = 0;
            this.CantidadReservado = 0;
            this.CantidadFaltante = 0;
            this.CantidadAtendido = 0;
            this.Enlazado = false;
            this.Atendido = false;
            this.NomProducto = String.Empty;
            this.CategoriaId = 0;
            this.CodigoUM = String.Empty;
            this.Stock = 0;
            this.TarifaId = 0;
            this.MonedaId = 0;
            this.Precio = 0;
        }

        public Int32 OrdenPedidoDetalleId { get; set; }
        public Int32 OrdenPedidoId { get; set; }
        public Int32 MercaderiaId { get; set; }
        public Int32 UnidadMedidaId { get; set; }
        public Decimal CantidadSolicitado { get; set; }
        public Decimal CantidadReservado { get; set; }
        public Decimal CantidadFaltante { get; set; }
        public Decimal CantidadAtendido { get; set; }
        public Boolean Enlazado { get; set; }
        public Boolean Atendido { get; set; }

        public Int32 TarifaId { get; set; }
        public Int32 MonedaId { get; set; }
        public Decimal Precio { get; set; }

    }
}
